plugins/modules/device_credential_intent.py

```diff
@@ -793,15 +863,20 @@ def get_global_credentials_params(self):
             self - The current object details.
 
         Returns:
-            global_credentials (dict) - All global device credentials details
+            global_credentials (dict) - All global device credentials details.
         """
 
-        global_credentials = self.dnac._exec(
-            family="discovery",
-            function='get_all_global_credentials_v2',
-        )
-        global_credentials = global_credentials.get("response")
-        self.log("All Global Device Credentials Details " + str(global_credentials))
+        try:
+            global_credentials = self.dnac._exec(
+                family="discovery",
+                function='get_all_global_credentials_v2',
+            )
+            global_credentials = global_credentials.get("response")
+            self.log("All Global Device Credentials Details " + str(global_credentials))
+        except Exception as exec:
+            self.log("Error while getting global device credentials: " + str(exec))
```

Comment:
don't we need to raise an exception here?

Reply:
we raised an exception right? :)